controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/HealthCheckTask.java

```diff
@@ -0,0 +1,166 @@
+/*
+ *  Copyright 2017 Pinterest, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.pinterest.rocksplicator.controller.tasks;
+
+import com.pinterest.rocksplicator.controller.bean.ClusterBean;
+import com.pinterest.rocksplicator.controller.bean.HostBean;
+import com.pinterest.rocksplicator.controller.bean.SegmentBean;
+import com.pinterest.rocksplicator.controller.bean.ShardBean;
+import com.pinterest.rocksplicator.controller.config.ConfigParser;
+import com.pinterest.rocksplicator.controller.util.AdminClientFactory;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.locks.InterProcessLock;
+import org.apache.curator.framework.recipes.locks.InterProcessMutex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+
+
+/**
+ * @author Ang Xu (angxu@pinterest.com)
+ */
+public class HealthCheckTask extends TaskBase<HealthCheckTask.Param> {
+
+  public static final Logger LOG = LoggerFactory.getLogger(HealthCheckTask.class);
+
+  @Inject
+  public CuratorFramework zkClient;
+
+  @Inject
+  public AdminClientFactory clientFactory;
+
+  public HealthCheckTask(Param param) {
+    super(param);
+  }
+
+  @Override
+  public void process(Context ctx) throws Exception {
+    final String clusterName = getParameter().getClusterName();
+    final String zkPath = getParameter().getZkPath();
+    final InterProcessLock lock =
```

Comment:
i think we decided to use mysql task queue lock instead of zk lock. @newpoo to confirm